bootstrapvz/plugins/puppet/tasks.py

```diff
@@ -36,28 +55,82 @@ def run(cls, info):
             raise TaskError(msg)
 
 
-class AddPackages(Task):
-    description = 'Add puppet package'
-    phase = phases.preparation
+class InstallPuppetlabsPC1RepoPackage(Task):
```

Comment:
Aha, now I understand your troubles with the `apt-get update` thing etc.  
All this package does is install the puppetlabs repo, which is a trusted gpg key and a file in `/etc/apt/sources.list.d`. You should instead do that via tasks, check out the apt module in `common/tasks` for how sources are added and keys are trusted (`AddBackports` and `InstallTrustedKeys`).